collect_app/src/main/java/org/odk/collect/android/provider/FormsProvider.java

```diff
@@ -200,19 +203,19 @@ public synchronized Uri insert(@NonNull Uri uri, ContentValues initialValues) {
             values.put(FormsColumns.MD5_HASH, md5);
 
             if (!values.containsKey(FormsColumns.JRCACHE_FILE_PATH)) {
-                String cachePath = Collect.CACHE_PATH + File.separator + md5
+                String cachePath = storagePathProvider.getDirPath(StorageSubdirectory.CACHE) + File.separator + md5
```

Comment:
This should be a db path, right?

Reply:
Ohh yes generaly that line was redundant at all because it was just:
```java
String cachePath = storagePathProvider.getDirPath(StorageSubdirectory.CACHE) + File.separator + md5 + ".formdef";
values.put(FormsColumns.JRCACHE_FILE_PATH, storagePathProvider.getCacheDbPath(cachePath));
```

so first I an absolute path was created and then used in getCacheDbPath() which deals with relative paths as well.
That means we can just use:

```java
values.put(FormsColumns.JRCACHE_FILE_PATH, storagePathProvider.getCacheDbPath(md5 + ".formdef"));
```

Fixed.